File: backend/schemas/carteirinha.py
from datetime import datetime, date
from typing import Optional, List
from uuid import UUID
from pydantic import BaseModel

class CarteirinhaBase(BaseModel):
    """Modelo base para Carteirinha"""
    paciente_id: UUID
    plano_saude_id: UUID
    numero_carteirinha: str
    data_validade: Optional[date] = None
    status: str = "ativa"
    motivo_inativacao: Optional[str] = None
    historico_status: Optional[List[dict]] = None

class CarteirinhaCreate(CarteirinhaBase):
    pass

class CarteirinhaUpdate(CarteirinhaBase):
    paciente_id: Optional[UUID] = None
    plano_saude_id: Optional[UUID] = None
    numero_carteirinha: Optional[str] = None
    data_validade: Optional[date] = None
    status: Optional[str] = None
    motivo_inativacao: Optional[str] = None
    historico_status: Optional[List[dict]] = None

class Carteirinha(CarteirinhaBase):
    """Modelo para a entidade Carteirinha"""
    id: UUID
    created_at: datetime
    updated_at: datetime
    created_by: Optional[UUID] = None
    updated_by: Optional[UUID] = None

    class Config:
        from_attributes = True

File: frontend/src/types/carteirinha.ts
export type CarteirinhaStatus = "ativa" | "inativa" | "suspensa" | "vencida";

export interface Carteirinha {
    id: string;
    paciente_id: string;
    paciente_nome?: string;
    plano_saude_id: string;
    plano_saude_nome?: string;
    numero_carteirinha: string;
    data_validade?: string;
    status: CarteirinhaStatus;
    motivo_inativacao?: string;
    historico_status?: any[];
    created_at: string;
    updated_at: string;
    created_by?: string;
    updated_by?: string;
    deleted_at?: string;
}

export interface CarteirinhaFormData {
    paciente_id: string;
    plano_saude_id: string;
    numero_carteirinha: string;
    data_validade?: string;
    status: CarteirinhaStatus;
    motivo_inativacao?: string;
    historico_status?: any[];
}

export interface CarteirinhaFilters {
    search?: string;
    paciente_id?: string;
}